fe/fe-core/src/main/java/com/starrocks/planner/AggregationNode.java

```diff
@@ -248,6 +249,8 @@ protected void toThrift(TPlanNode msg) {
             msg.agg_node.setStreaming_preaggregation_mode(TStreamingPreaggregationMode.FORCE_STREAMING);
         } else if (streamingPreaggregationMode.equalsIgnoreCase(FORCE_PREAGGREGATION)) {
             msg.agg_node.setStreaming_preaggregation_mode(TStreamingPreaggregationMode.FORCE_PREAGGREGATION);
+        } else if (streamingPreaggregationMode.equalsIgnoreCase(LIMITED)) {
```

Comment:
Whether adopting LIMITED or not, should be determined by  FE,  for an example: the number of group-by columns exceeds a specified threashold or there exists some high-cardinality group-by columns, FE adopts LIMITED mode.

Reply:
This parameter is just for testing and validating the effect or avoiding bad cases, and should not be set by the actual user. In BE this parameter is set by spill.

Reply:
```
void AggregateStreamingSinkOperator::set_execute_mode(int performance_level) {
    if (_aggregator->streaming_preaggregation_mode() == TStreamingPreaggregationMode::AUTO) {
        _aggregator->streaming_preaggregation_mode() = TStreamingPreaggregationMode::LIMITED_MEM;
    }
    _limited_mem_state.limited_memory_size = _aggregator->hash_map_memory_usage();
}
```